sourcemap/store.go

```diff
@@ -44,39 +41,103 @@ var (
 
 // Store holds information necessary to fetch a sourcemap, either from an Elasticsearch instance or an internal cache.
 type Store struct {
-	cache   *gocache.Cache
-	esStore *esStore
-	logger  *logp.Logger
+	cache       *gocache.Cache
+	backend     backend
+	logger      *logp.Logger
+	waiters     chan struct{}
+	waitTimeout time.Duration
+
+	sync.Mutex
```

Comment:
```suggestion
	mu sync.Mutex
```

so we don't expose the mutex to the world

Reply:
:100: